webplugin/js/app/km-rich-text-event-handler.js

```diff
@@ -654,8 +657,19 @@ Kommunicate.richMsgEventHandler = {
         }
         var messagePxy = {};
         var msgMetadata = {};
-        replyText && (messagePxy.message = replyText); //message to send
-
+        var postFormDataAsMessage =
+            ispostBackAsMessage &&
+            data &&
```

Comment:
can you share a sample for how the data looks like? is it an array?
couldn't understand why data need to be converted to JSON first and then remove the JSON attributes from it?

Reply:
![image](https://user-images.githubusercontent.com/22856546/133040527-31aca31b-731c-4e29-82c8-d704431b3b60.png)

Reply:
directly convert object to the format you need, json stringify is not required